### ISSUE TITLE
Unspecified attribute which is used as a pk in a GSI is set to empty string on expression

### ISSUE THREAD

**Issue by PaulJNewell77** [tywalch/electrodb#229]
**Describe the bug**
For a table with a GSI which has pk/sk that map to non-required attributes (i.e a sparse index), calling put with an item that does not have the pk attribute specified but does have the sk attribute specified, ElectrobDb adds the pk attribute with an empty string value to the expression, which results in the error below.

The first Playground link below demonstrates this symptom. Note that `code` is not specified on the put object, but is set to empty string in the expression. The second Playground link shows that this does not happen if the sk attribute `task` is not set.

**ElectroDB Version**
2.5.1

**ElectroDB Playground Link**
[ElectroDB Playground 1](https://electrodb.fun/?#code/JYWwDg9gTgLgBAbzgUQHY2DAngGjgZQFMoA3YAY0LgF84AzKCEOAIkIBtDyZGATAIxYBuAFAjyEVAGd4MAIb9OcALyssEAK5QA+vMWFtqOSELCxE6bLlSA1lJVxUhAO4p0mLAAoRcRD99wIBC8HABcfgEBhO7Y4SzytlIsOP6RJMRSwJJxAIzJqQFSxGSUcQk2cmBgLAXUKZFyMDzA-BowhFLhCAW+7cZdPQHYYIRxMlDAqADm+ZGRUIQAjhrAC7zhPBqEg3WD5QNzQ1gjY83Ts4dwC8urhOv0cuxFO-VzEiEHl8OjrOOTM69DtcVmtwnRHkVAQE6pFalDJiEAB4dT6RMCMABWXBgnQilzANlRlzowA49xYBIulzgEnAEEy7XCAG14oRjCwALqDXy7F5iQ7OTAACwAwsEft1qQB6KVwKRgORQIpwBGERFwGBCxpyoWadi8OCSdhYOD8KiQMAadiNO4qug08Vywgwbkq1BIuKCzUAWnehG9qsRVLmBKJhxJZLifuDh1pkAZPxZ0a51N21NsYbmEf1ZWsNhjbyY8cwifiec5ruoO0OtX8acQGoUSirAEpRCIZXAACIQf5wS3wL1Cw38LHcOBDzTwP1XDpW+CTOCEcDYOVnKYOkJu-uK4xSETlKQAOgHniQfRA4QA5Ber3h9nAb3mr9QW0ephBPG2REA)

[ElectroDB Playground 2](https://electrodb.fun/?#code/JYWwDg9gTgLgBAbzgUQHY2DAngGjgZQFMoA3YAY0LgF84AzKCEOAIkIBtDyZGATAIxYBuAFAjyEVAGd4MAIb9OcALyssEAK5QA+vMWFtqOSELCxE6bLlSA1lJVxUhAO4p0mLAAoRcRD99wIBC8HABcfgEBhO7Y4SzytlIsOP6RJMRSwJJxAIzJqQFSxGSUcQk2cmBgLAXUKZFyMDzA-BowhFLhCAW+7cZdPQHYYIRxMlDAqADm+ZGRUIQAjhrAC7zhPBqEg3WD5QNzQ1gjY83Ts4dwC8urhOv0cuxFO-VzgxIhB5fDo6zjkzNXodris1uE6I8ikCAnVIrVoZMQgAPDpfSJgRgAKy4ME6EUuYBsaMudGAHHuLEJF0ucAk4AgmXa4QA2vFCMYWABdQa+XYvMSHZyYAAWAGFgr9ujSAPTSuBSMByKBFOCIwhIuAwYWNeXCzTsXhwSTsLBwfhUSBgDTsRp3VV0WkS+WEGA81WoZFxIVagC0H0IPrVSOpc0JxMOpPJcX9IcOdMgjN+rJj3JpuxptnDc0jBrK1hssbm8YZmCT8XzXLd1B2h1q-nTiE1CiU1YAlKIRLK4AARCAAuBW+De4VG-jY7hwYeaeD+q4da3wSZwQjgbDys5TR0hd0DpXGKQicpSAB0g88SD6IHCAHJL9e8Ps4BoPYRSU5eNRW8ephBPO2REAA)

**Entity/Service Definitions**
As in Playground

**Expected behavior**
If an attribute is not set on the object in the put request it should not be present in the expression

**Errors**
```
One or more parameter values are not valid. A value specified for a secondary index key is not supported. The AttributeValue for a key attribute cannot contain an empty string value
```

**Comment by tywalch** [comment]
Thanks Paul! I think this one should actually be a pretty easy one -- I'll hopefully have updates soon

**Comment by tywalch** [comment]
I have addressed this issue in version `2.8.2`, let me know if that fixes your issue 👍

**Comment by tywalch** [comment]
Feel free to reopen this issue if the issue wasn't resolved with `2.8.2`
